### ISSUE TITLE
Function call in INITIAL block of NET_RECEIVE result into compilation error

### ISSUE THREAD

**Issue by pramodk** [BlueBrain/mod2c#39]
I am testing a mod file from Michele ([fin.mod](https://github.com/pramodk/modeldb-bulb3d-sim/blob/master/sim/fi.mod)):

```
DERIVATIVE state {
    A' = -A/tau1
    B' = -B/tau2
}

FUNCTION plast(step) {
    plast = (1 - 1/(1 + exp((step - sighalf)/sigslope)))^sigexp

}

NET_RECEIVE(weight, s, w, tlast (ms)) {
    INITIAL {
        :s = 0

                if(s == 0) {
                    w = 0
                } else {
                    w = weight*plast(s)
                }

        tlast = -1e9(ms)
    }
    : some other code
}
```

and compiling this gives:

```
 -> Compiling x86_64/core/mod2c/fi.cpp
x86_64/core/mod2c/fi.cpp(560): error: identifier "v" is undefined
       _args[2] = _args[0] * plast ( _threadargscomma_ _args[1] ) ;
```

This is because `net_init` function gets defined as:

```
void _net_init(Point_process* _pnt, int _weight_index, double _lflag) {

   double* _p; Datum* _ppvar; ThreadDatum* _thread;
   Memb_list* _ml; int _cntml_padded, _cntml_actual; int _iml; double* _args;

   NrnThread* _nt;
   int _tid = _pnt->_tid;
   _nt = nrn_threads + _tid;
   _thread = (ThreadDatum*)0;
   double *_weights = _nt->_weights;
   _args = _weights + _weight_index;
   _ml = _nt->_ml_list[_pnt->_type];
   _cntml_actual = _ml->_nodecount;
   _cntml_padded = _ml->_nodecount_padded;
   _iml = _pnt->_i_instance;
#if LAYOUT == 1 /*AoS*/
   _p = _ml->_data + _iml*_psize; _ppvar = _ml->_pdata + _iml*_ppsize;
#endif
#if LAYOUT == 0 /*SoA*/
   _p = _ml->_data; _ppvar = _ml->_pdata;
#endif
#if LAYOUT > 1 /*AoSoA*/
#error AoSoA not implemented.
#endif
 if ( _args[1]  == 0.0 ) {
     _args[2] = 0.0 ;
     }
   else {
     _args[2] = _args[0] * plast ( _threadargscomma_ _args[1] ) ;
     }
   _args[3] = - 1e9 ;
   }
```

and `_threadargscomma_` is defined as : 

```
#define _threadargscomma_ _iml, _cntml_padded, _p, _ppvar, _thread, _nt, v,
```

@nrnhines : as `v` is not defined, could you see what is best way to resolve this?

In case of NEURON, I see:

```
#define _threadargscomma_ _p, _ppvar, _thread, _nt,
```

i.e. `v` is not passed as an argument and hence not an issue.

**Comment by nrnhines** [comment]
Looking back to 2008, I see that in nrn/src/nmodl/nocpout.c ,```_threadargscommna__``` has never had v as an arg. And it is not in mod2c/src/mod2c/nocpout.c. It was added in mod2c_core for
```
commit 0ec6ff42b9ffeef43fe8ab6b78d5143930e81783
Author: Michael Hines <michael.hines@yale.edu>
Date:   Tue Feb 10 08:35:32 2015 +0100

    v has become a part of _threadargs_ in order to reduce the data size.
```
As that changeset is not too large, I think there is a good chance it can be effectively reverted. I'm guessing (hoping) there is no place in VERBATIM mod files where
the expansion was used  instead of ```_threadargs..._ ```

**Comment by pramodk** [comment]
fixed by #40
